File: examples/python/hello_prompt.py
import argparse
from pathlib import Path
import sys

parser = argparse.ArgumentParser(description="Demo prompt script")
parser.add_argument("text")
parser.add_argument("file_path")
args = parser.parse_args()

print("[hello] Hello from Python!")
print(f"[info] Text argument : {args.text}")
print(f"[info] File argument : {args.file_path}")

file_path = Path(args.file_path)
if not file_path.exists():
    print(f"File '{file_path}' does not exist.", file=sys.stderr)
    sys.exit(1)

print("QUESTION: Display the file content? (Yes/No/Maybe)")
answer = sys.stdin.readline().strip().upper()

if answer == "YES":
    print("[content]")
    for line in file_path.read_text().splitlines():
        print(f"  {line}")
elif answer == "NO":
    print("[skip] Content display skipped.")
elif answer == "MAYBE":
    print("MAYBE_SELECTED - user could not decide.")
    sys.exit(2)
else:
    print(f"[warn] Unexpected answer '{answer}'. Treating as NO.")

File: examples/node/hello_prompt.ts
import { readFileSync, existsSync } from "fs";

const [textArg, filePath] = process.argv.slice(2);
if (!textArg || !filePath) {
  console.error("Usage: ts-node hello_prompt.ts <text> <filePath>");
  process.exit(1);
}

console.log("[hello] Hello from TypeScript!");
console.log(`[info] Text argument : ${textArg}`);
console.log(`[info] File argument : ${filePath}`);

if (!existsSync(filePath)) {
  console.error(`File '${filePath}' does not exist.`);
  process.exit(1);
}

console.log("QUESTION: Display the file content? (Yes/No/Maybe)");

process.stdin.setEncoding("utf-8");
process.stdin.once("data", (chunk) => {
  const answer = chunk.trim().toUpperCase();
  if (answer === "YES") {
    console.log("[content]");
    readFileSync(filePath, "utf-8")
      .split(/\r?\n/)
      .forEach((line) => console.log(`  ${line}`));
       process.exit(0);
  } else if (answer === "NO") {
    console.log("[skip] Content display skipped.");
    process.exit(1);
  } else if (answer === "MAYBE") {
    console.log("MAYBE_SELECTED - user could not decide.");
    process.exit(0);
  } else {
    console.log(`[warn] Unexpected answer '${answer}'. Treating as NO.`);
  }
});
